magic.js

```diff
@@ -1221,6 +1221,41 @@ function uid(sec, cb) {
   }).catch((err) => { return done(err); })
 }
 
+/**
+ * Provides timing safe comparisons of two strings to prevent
+ * timing based attacks. Returns true if the strings are the
+ * same and false if not.
+ * @function
+ * @api public
+ * 
+ * @param {string} input - The first string to check
+ * @param {string} ref - The reference string to check against
+ * @returns {Boolean} - true if they match, false otherwise
+ */
+module.exports.util.timingSafeCompare = (input, ref) => {
+  inputIsString = typeof input === 'string' || input instanceof String;
+  refIsString = typeof ref === 'string' || ref instanceof String;
+  if (!inputIsString || !refIsString) throw new TypeError('Inputs must be Strings');
+  
+  let inputLength = Buffer.byteLength(input);
+  let refLength = Buffer.byteLength(ref);
+
+  /*  Allocate two buffers, making the input buffer the length. 
+  Continue the comparison of the two buffers 
+  with the length evaluation failing at the end to not give
+  away the length of the reference string.  
+  */
+  let inputBuffer = Buffer.alloc(inputLength);
+  inputBuffer.write(input, 0, inputLength);
```

Comment:
There's no need for the second and third arguments. These are the default ones.

Reply:
I personally like the explicit args rather than "hidden" defaults. Plus, it matches the line further down when we use `inputLength` instead of `refLength`.

Reply:
Either is fine with me 👍 Because the logic is a little strange compared to normal functions I decided to go with the more explicit approach but if it is more readable I can use the default arguments.